js/src/Tooltip.jsx

```diff
@@ -49,90 +45,64 @@ var typeLabelStyle = {
   'border-radius': '2px',
   'font-size': '14px',
   'text-align': 'right',
-  'padding': '0px 5px',
+  'padding': '0px 5px'
 }
 
-function decompartmentalizeCheck (id, type) {
+class DefaultTooltip extends Component {
+  decompartmentalizeCheck (id, type) {
   // ID without compartment, if metabolite.
-  return type === 'metabolite' ? utils.decompartmentalize(id)[0] : id
-
-}
-
-function capitalizeFirstLetter (s) {
-  return s === null ? s : s.charAt(0).toUpperCase() + s.slice(1)
-}
-
-// Create the component
-var DefaultTooltip = createComponent({
-  displayName: 'DefaultTooltip',
-
-  init: function () {
-    return {
-      biggId: '',
-      name: '',
-      loc: { x: 0, y: 0 },
-      data: null,
-      type: null,
+    return type === 'metabolite'
+    ? utils.decompartmentalize(id)[0]
+    : id
+  }
+
+  openBigg () {
+    let type = this.props.type
+    let biggId = this.props.biggId
+    let pref = 'http://bigg.ucsd.edu/'
+    let url = (type === 'gene'
+              ? pref + 'search?query=' + biggId
+              : pref + 'universal/' + type + 's/' + this.decompartmentalizeCheck(biggId, type))
+    window.open(url)
+  }
+
+  capitalizeFirstLetter (s) {
+    if (s !== undefined) {
```

Comment:
This will return `undefined`, same for `null` it will return `null`.  How about empty string?
I think if silent fail should be avoided, 
```
capitalizeFirstLetter(s) {
  return s && s.charAt(0).toUpperCase() + s.slice(1)
}
```

Reply:
Changed the function so that if 's' is a string it capitalizes the first letter otherwise it sends a warning to the console.